command/registry.go

```diff
@@ -14,35 +121,141 @@ import (
 // Factory is a function that returns a new instance of a CLI-sub command.
 type Factory func(cli.Ui) (cli.Command, error)
 
-// Register adds a new CLI sub-command to the registry.
-func Register(name string, fn Factory) {
-	if registry == nil {
-		registry = make(map[string]Factory)
-	}
+// Entry is a function that returns a command's name and a Factory for that command.
+type Entry func() (string, Factory)
```

Comment:
Can you see if `Factory` and `Entry` can be unexported? Not sure if they're used outside this package

Reply:
Also, I think if we make `Entry` a struct instead we can simply pass around structs than funcs

Reply:
oh ya, a struct makes a lot more sense. I don’t know what I was thinking 😅